### DISCUSSION TITLE
Hands-On Machine Learning (3rd ed)

### DISCUSSION THREAD

**Discussion by chengjun** [computational-class/css#59] [Show and tell]
# Hands-On Machine Learning with Scikit-Learn, Keras, and TensorFlow (3rd ed)

## Machine Learning Notebooks
Welcome to the Machine Learning Notebooks!

![image](https://user-images.githubusercontent.com/543384/226279680-4dd4a3db-eec5-46c7-966a-462861d7e593.png)

### Code Examples

All the code examples in this book are open source and available online at https://github.com/ageron/handson-ml3, as Jupyter notebooks.
- These are interactive documents containing text, images, and executable code snippets (Python in our case). 
- The easiest and quickest way to get started is to run these notebooks using Google Colab.

[Prerequisites](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/index.ipynb#Prerequisites) (see below)

## Notebooks

- [The Machine Learning landscape](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/01_the_machine_learning_landscape.ipynb)
- [End-to-end Machine Learning project](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/02_end_to_end_machine_learning_project.ipynb)
- [Classification](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/03_classification.ipynb)
- [Training Models](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/04_training_linear_models.ipynb)
- [Support Vector Machines](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/05_support_vector_machines.ipynb)
- [Decision Trees](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/06_decision_trees.ipynb)
- [Ensemble Learning and Random Forests](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/07_ensemble_learning_and_random_forests.ipynb)
- [Dimensionality Reduction](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/08_dimensionality_reduction.ipynb)
- [Unsupervised Learning Techniques](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/09_unsupervised_learning.ipynb)
- [Artificial Neural Nets with Keras](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/10_neural_nets_with_keras.ipynb)
- [Training Deep Neural Networks](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/11_training_deep_neural_networks.ipynb)
- [Custom Models and Training with TensorFlow](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/12_custom_models_and_training_with_tensorflow.ipynb)
- [Loading and Preprocessing Data](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/13_loading_and_preprocessing_data.ipynb)
- [Deep Computer Vision Using Convolutional Neural Networks](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/14_deep_computer_vision_with_cnns.ipynb)
- [Processing Sequences Using RNNs and CNNs](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/15_processing_sequences_using_rnns_and_cnns.ipynb)
- [Natural Language Processing with RNNs and Attention](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/16_nlp_with_rnns_and_attention.ipynb)
- [Autoencoders, GANs, and Diffusion Models](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/17_autoencoders_gans_and_diffusion_models.ipynb)
- [Reinforcement Learning](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/18_reinforcement_learning.ipynb)
- [Training and Deploying TensorFlow Models at Scale](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/19_training_and_deploying_at_scale.ipynb)

### Scientific Python tutorials
- [NumPy](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/tools_numpy.ipynb)
- [Matplotlib](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/tools_matplotlib.ipynb)
- [Pandas](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/tools_pandas.ipynb)

### Math Tutorials
[Linear Algebra](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/math_linear_algebra.ipynb)
[Differential Calculus](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/math_differential_calculus.ipynb)

### Extra Material
[Auto-differentiation](https://colab.research.google.com/github/ageron/handson-ml3/blob/main/extra_autodiff.ipynb)

### Misc.
[Equations](https://github.com/ageron/handson-ml3/blob/main/book_equations.pdf) (list of equations in the book)

### Prerequisites
To understand Python – you don't need to be an expert python programmer, but you do need to know the basics. If you don't, the official [Python tutorial](https://docs.python.org/3/tutorial/) is a good place to start.

Scientific Python – We will be using a few popular python libraries, in particular NumPy, matplotlib and pandas. If you are not familiar with these libraries, you should probably start by going through the tutorials in the Tools section (especially NumPy).

Math – We will also use some notions of Linear Algebra, Calculus, Statistics and Probability theory. You should be able to follow along if you learned these in the past as it won't be very advanced, but if you don't know about these topics or you need a refresher then go through the appropriate introduction in the Math section.

### To run the examples
Jupyter – These notebooks are based on Jupyter. You can run these notebooks in just one click using a hosted platform such as Binder, Deepnote or Colaboratory (no installation required), or you can just view them using Jupyter.org's viewer, or you can install everything on your machine, as you prefer. Check out the [home page](https://github.com/ageron/handson-ml3/) for more details.

**Comment by chengjun** [comment]
> [小玄子玩不转](https://www.douban.com/people/159771914/)  [2023-01-19 17:44:14](https://book.douban.com/comment/3653451765) 中国香港 第三版新增了lightgbm的基础算法 histogram-based gradient boosting ，几乎重写了RNN部分，引用了更多的第二版（19）以来出版的paper，甚至还补充了diffusion model。实践与理论操作最佳入门指南了

**Comment by chengjun** [comment]
另外一本更基础的书是《Python机器学习基础教程》

![image](https://user-images.githubusercontent.com/543384/226501817-7cf5920b-8f1a-4f1c-80ae-8b46eaacfa2f.png)

https://github.com/amueller/introduction_to_ml_with_python
